src/Polymesh.ts

```diff
@@ -193,32 +190,17 @@ export class Polymesh {
     } = params;
     let context: Context;
 
-    /* istanbul ignore next: part of configuration, doesn't need to be tested */
-    const wsp = new WebSocketAsPromised(nodeUrl, {
-      createWebSocket: url => (new W3CWebSocket(url) as unknown) as WebSocket,
-      packMessage: data => JSON.stringify(data),
-      unpackMessage: data => JSON.parse(data.toString()),
-      attachRequestId: (data, requestId) => Object.assign({ id: requestId }, data),
-      extractRequestId: data => data && data.id,
+    await assertExpectedChainVersion(nodeUrl).catch(error => {
+      if (
```

Comment:
I figured since I was making a change here, I might as well make it a warning.

https://polymathnetwork.slack.com/archives/C026L2FKT25/p1646310363628679?thread_ts=1646304939.113329&cid=C026L2FKT25